powersimdata/input/transform_profile.py

```diff
@@ -1,111 +1,109 @@
 import copy
 
-from powersimdata.input.grid import Grid
 from powersimdata.input.input_data import InputData
 
 
 class TransformProfile(object):
-    """Transforms profile according to operations listed in change table."""
+    """Transform profile according to operations listed in change table."""
 
     def __init__(self, scenario_info, grid, ct):
-        """Constructor
+        """Constructor.
 
         :param dict scenario_info: scenario information.
-        :param powersimdata.input.grid.Grid grid: a Grid object.
+        :param powersimdata.input.grid.Grid grid: a Grid object previously
+            transformed.
         :param dict ct: change table.
         """
         self._input_data = InputData()
         self.scenario_info = scenario_info
-        self.grid = copy.deepcopy(grid)
+
         self.ct = copy.deepcopy(ct)
+        self.grid = copy.deepcopy(grid)
+
         self.scale_keys = {
             "wind": {"wind", "wind_offshore"},
             "solar": {"solar"},
             "hydro": {"hydro"},
             "demand": {"demand"},
         }
-        self.n_new_plant = (
-            0 if "new_plant" not in self.ct.keys() else len(self.ct["new_plant"])
-        )
-        self.base_plant = Grid(self.grid.interconnect).plant
+        self.n_new_plant, self.n_new_clean_plant = self._get_number_of_new_plant()
+
+    def _get_number_of_new_plant(self):
+        """Return the total number of new plant and new clean plant.
+
+        :return: (*tuple*) -- first element is the total number of new plant and second
+            element is the total number of new clean plant (*hydro*, *solar*,
+            *onshore wind* and *offshore wind*).
```

Comment:
`new clean plant` -> `new plants with profiles`. Geothermal and nuclear are clean too but don't have profiles.

Reply:
Done